sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala

```diff
@@ -71,14 +71,41 @@ private class ChangeFeedBatch
 
         if (metadataLog.get(0).isDefined) {
           val offsetJson = metadataLog.get(0).get
-          ChangeFeedOffset.fromJson(offsetJson).changeFeedState
+
+          log.logDebug(s"Start offset retrieved from file location '$startOffsetLocation' for batchId: $batchId " +
+            s"-> offset: '$offsetJson'.")
+
+          val changeFeedStateBase64 = ChangeFeedOffset.fromJson(offsetJson).changeFeedState
+          val expectedContainerResourceId = container.read().block().getProperties.getResourceId
+          val offsetIsValid = SparkBridgeImplementationInternal.validateCollectionRidOfChangeFeedState(
+            changeFeedStateBase64,
+            expectedContainerResourceId,
+            changeFeedConfig.ignoreOffsetWhenInvalid
+          )
+
+          if (offsetIsValid) {
+            changeFeedStateBase64
+          } else {
+            val newOffsetJson = CosmosPartitionPlanner.createInitialOffset(
+              container, changeFeedConfig, partitioningConfig, None)
+            log.logDebug(s"Invalid Start offset retrieved from file location '$startOffsetLocation' for batchId: $batchId " +
```

Comment:
this could be info log I think - feels can be very useful

Reply:
I'll actually make it a warning